diff --git a/pom.xml b/pom.xml
index e793446..dac24a8 100644
--- a/pom.xml
+++ b/pom.xml
@@ -1,10 +1,6 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <parent>
         <groupId>codemining</groupId>
-        <artifactId>codemining-all</artifactId>
         <version>1.1-SNAPSHOT</version>
-        <relativePath>../pom.xml</relativePath>
-  </parent>
 
 
   <modelVersion>4.0.0</modelVersion>
diff --git a/src/main/java/codemining/ast/java/tui/MethodNamingDatasetCreator.java b/src/main/java/codemining/ast/java/tui/MethodNamingDatasetCreator.java
index 14a5310..8135d53 100644
--- a/src/main/java/codemining/ast/java/tui/MethodNamingDatasetCreator.java
+++ b/src/main/java/codemining/ast/java/tui/MethodNamingDatasetCreator.java
@@ -24,7 +24,7 @@ import com.google.gson.Gson;
 import codemining.ast.AstNodeSymbol;
 import codemining.ast.TreeNode;
 import codemining.ast.java.JavaAstTreeExtractor;
-import codemining.java.codeutils.MethodExtractor;
+import codemining.java.codedata.MethodRetriever;
 import codemining.java.tokenizers.JavaTokenizer;
 import codemining.util.data.Pair;
 
@@ -76,7 +76,7 @@ public class MethodNamingDatasetCreator {
 
 	private static Stream<MethodNameWithAst> getMethods(File file, File inputFolder) {
 		try {
-			return MethodExtractor.getMethods(file).stream()
+			return MethodRetriever.getMethodNodes(file).values().stream()
 					.filter(m -> m != null && m.getBody() != null && !m.isConstructor()).map(m -> {
 						final Tree tokens = getMethodAst(m);
 						return new MethodNameWithAst(file.toString().substring(inputFolder.toString().length()),
diff --git a/src/main/java/codemining/lm/tsg/tui/java/SampleBlockedTSG.java b/src/main/java/codemining/lm/tsg/tui/java/SampleBlockedTSG.java
index 7af8bc5..648b13a 100644
--- a/src/main/java/codemining/lm/tsg/tui/java/SampleBlockedTSG.java
+++ b/src/main/java/codemining/lm/tsg/tui/java/SampleBlockedTSG.java
@@ -19,7 +19,7 @@ import codemining.ast.java.BinaryJavaAstTreeExtractor;
 import codemining.ast.java.DelegatedVariableTypeJavaTreeExtractor;
 import codemining.ast.java.JavaAstTreeExtractor;
 import codemining.ast.java.VariableTypeJavaTreeExtractor;
-import codemining.java.codeutils.MethodExtractor;
+import codemining.java.codedata.MethodRetriever;
 import codemining.java.tokenizers.JavaTokenizer;
 import codemining.lm.tsg.FormattedTSGrammar;
 import codemining.lm.tsg.TSGNode;
@@ -123,7 +123,7 @@ public class SampleBlockedTSG {
                         nNodes += ast.getTreeSize();
                         filter.addTree(ast);
                     } else {
-                        for (final MethodDeclaration method : MethodExtractor.getMethods(fi)) {
+                        for (final MethodDeclaration method : MethodRetriever.getMethodNodes(fi).values()) {
                             if (method.getBody() == null) {
                                 continue;
                             }
